s3/src/main/java/ch/cyberduck/core/s3/S3Session.java

```diff
@@ -335,25 +342,45 @@ public void login(final LoginCallback prompt, final CancelCallback cancel) throw
             log.warn(String.format("Skip verifying credentials with previous successful authentication event for %s", this));
             return;
         }
-        try {
-            final Path home = new DelegatingHomeFeature(new DefaultPathHomeFeature(host)).find();
-            final Location.Name location = new S3LocationFeature(S3Session.this, regions).getLocation(home);
-            if(log.isDebugEnabled()) {
-                log.debug(String.format("Retrieved region %s", location));
-            }
-            if(!Location.unknown.equals(location)) {
+        if(S3Session.isAwsHostname(host.getHostname(), false)) {
+            final CustomClientConfiguration configuration = new CustomClientConfiguration(host,
+                    new ThreadLocalHostnameDelegatingTrustManager(trust, host.getHostname()), key);
+            final AWSSecurityTokenServiceClientBuilder builder = AWSSecurityTokenServiceClientBuilder.standard()
+                    .withCredentials(AWSCredentialsConfigurator.toAWSCredentialsProvider(client.getProviderCredentials()))
+                    .withClientConfiguration(configuration);
+            final AWSSecurityTokenService service = builder.build();
+            // Returns details about the IAM user or role whose credentials are used to call the operation.
+            // No permissions are required to perform this operation.
+            try {
+                final GetCallerIdentityResult identity = service.getCallerIdentity(new GetCallerIdentityRequest());
                 if(log.isDebugEnabled()) {
-                    log.debug(String.format("Set default region to %s determined from %s", location, home));
+                    log.debug(String.format("Successfully verified credentials for %s", identity));
                 }
-                //
-                host.setProperty("s3.location", location.getIdentifier());
+            }
+            catch(AWSSecurityTokenServiceException e) {
+                throw new STSExceptionMappingService().map(e);
             }
         }
-        catch(AccessDeniedException | InteroperabilityException e) {
-            log.warn(String.format("Failure %s querying region", e));
-            final Path home = new DefaultHomeFinderService(this).find();
-            if(log.isDebugEnabled()) {
-                log.debug(String.format("Retrieved %s", home));
+        else {
+            final Path home = new DelegatingHomeFeature(new DefaultPathHomeFeature(host)).find();
+            if(home.isRoot()) {
+                if(log.isDebugEnabled()) {
+                    log.debug(String.format("Skip querying region for %s", home));
+                }
+                new S3ListService(this, acl).list(home, new DisabledListProgressListener());
+            }
+            else {
+                final Location.Name location = new S3LocationFeature(this, regions).getLocation(home);
```

Comment:
Why can the exception handling for `AccessDeniedException | InteroperabilityException` be omitted? What happens in this case now?

Reply:
See commit c78789d7a3ec2f468ff533dbc6c3e82fb35ec5cd

Reply:
At least the Spectra tests are red - not clear if this is related to this specific change or another one in this PR.

Reply:
Spectra integration tests are ✅